dev/ToastNotifications/NotificationProperties.cpp

```diff
@@ -0,0 +1,148 @@
+{
+    auto lock = m_lock.lock_shared();
+
+    *notificationId = m_notificationId;
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Payload(_Out_ unsigned int* payloadSize, _Out_ byte** payload) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    size_t tempPayloadSize = m_payload.size();
+
+    auto tempPayload = wil::unique_cotaskmem_array_ptr<byte>(static_cast<byte*>(CoTaskMemAlloc(tempPayloadSize)), tempPayloadSize);
+    CopyMemory(tempPayload.data(), m_payload.get(), tempPayloadSize);
+
+    *payloadSize = static_cast<unsigned int>(tempPayloadSize);
+    *payload = tempPayload.release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_PayloadSize(_Out_ unsigned int* payloadSize) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *payloadSize = static_cast<unsigned int>(m_payload.size());
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Tag(_Out_ HSTRING* tag) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    *tag = safe_make_unique_hstring(m_tag.c_str()).release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Group(_Out_ HSTRING* group) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    *group = safe_make_unique_hstring(m_group.c_str()).release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Expiry(_Out_ unsigned long long* expiry) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *expiry = static_cast<ULONGLONG>(m_expiry.dwHighDateTime) << 32 | static_cast<ULONGLONG>(m_expiry.dwLowDateTime);
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_ArrivalTime(_Out_ unsigned long long* arrivalTime) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *arrivalTime = static_cast<ULONGLONG>(m_arrivalTime.dwHighDateTime) << 32 | static_cast<ULONGLONG>(m_arrivalTime.dwLowDateTime);
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_BootId(_Out_ unsigned long long* bootId) noexcept
+{
+    *bootId = 0;
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_ExpiresOnReboot(_Out_ boolean* expiresOnReboot) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *expiresOnReboot = m_expiresOnReboot;
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_ToastProgressData(_Out_ ToastABI::IToastProgressData** progressData) noexcept
+{
+    // TODO: Uncomment once ToastProgressData is implemented
+    // auto lock = m_lock.lock_shared();
+    // *toastProgressData = m_toastProgressData.get();
+    *progressData = nullptr;
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_ActivityId(_Out_ GUID* activityId) noexcept
+{
+    *activityId = GUID_NULL;
```

Comment:
Same feedback here. Declare m_activityId. Set that in the ctor. Its ok if it is GUID_NULL when being set.

Reply:
There is CV in RS5. I don't think we need activityId for anything. Ideally, I should have noticed this and not even declare it as part of the interface declaration.

If we ever need this, we can take it separately. The UDK is ready to extract whatever GUID comes from this object.